homeassistant/components/geniushub/__init__.py

```diff
@@ -57,17 +66,20 @@
 V1_API_SCHEMA = vol.Schema(
     {
         vol.Required(CONF_TOKEN): cv.string,
-        vol.Required(CONF_MAC): vol.Match(MAC_ADDRESS_REGEXP),
+        vol.Required(CONF_MAC): cv.string,  # vol.Match(MAC_ADDRESS_REGEXP),
     }
 )
```

Comment:
Oh I now also remember, we talked about what if we can use the cloud configuration to connect and get all devices eagerly. So if you have for example 5 devices, you don't have to set up your token 5 times

Reply:
The GH is responsible for returning a collection of devices. Don't believe the majority of users would have more than one GH. How many properties do you know that have more than one NEST controller?

Reply:
But then we can lose the requirement of the MAC, which is more user friendly

Reply:
We can't drop the need for a MAC address as this is required by the underlying service.

Reply:
Let me rephrase, I think I asked but i forgot.

Can we ask Geniushub a list of all the devices? If so, we can just setup all those devices. this avoids the need of setting up a cloud entry for every single device you have.

Reply:
Yes. The Genius Hub is the gateway to the devices in that system. It returns a collection of devices including TRVs, room sensors (temperature, light level, presence), switches. These devices are represented in HA as entities. I personally believe it would be better to represent these as devices as there are physical devices (perhaps something for a future PR). 

You cannot connect to individual device except via the Genius Hub.

The last point isn't strictly true and the Genius Hub network is essentially, I believe, a z-wave or zigbee network.

The TRVs and sensors are presented to HA as climate controls and also as sensors. For example, a room may have two radiators (two TRVs) and one room sensor. This would appear in HA as four entities. One for each device (two TRVs and a room sensor) plus a climate entity.

Reply:
I am not talking about entities yet, I am talking about config entries. So that would mean if I have 5 devices, for the cloud intergation I would be left with 5 config entries. Do I understand that correctly?

Reply:
A single config entry is made to log into the hub.

Reply:
That doesn't answer the question though. If I have 2 hubs, do I have to set it up twice?

Reply:
Two hubs will need to be setup twice. Once for each hub.

I believe this is the same behaviour as the app for the Genius hub but I am not sure as I only have one hub.

Reply:
Yes, but where is that limitation? I want to try to avoid having the user to get the mac address if that is not needed.

Reply:
Reading the [documentation](https://www.home-assistant.io/integrations/geniushub/), you will see how the MAC address is used. 

It is possible to connect to the same hub locally and via the cloud. To tie the two instances together in HA the MAC address of the hub is required. If the MAC address for the cloud is different then two instances of the same hub will appear in HA. Feature or limitation?

IMO, most people will connect locally where the MAC address isn't required. I see no reason why anyone, other than for remote access beyond HA would use the cloud service.

This limitation has always been there with this integration. What you keep asking is going well beyond the scope of this PR.

Reply:
I just try to do some due dilligence to know if this is the right path to take. Changing this after this PR is practically impossible, hence I am asking now. Like I only want to know the full context to have a good view about how its going to look like. But my question is not answered, does the user need to fill in a mac address, or can we fetch a list of all the hubs. Because if the reality is "we fetch all the hubs, but you need to pick the one with this macaddress" we might as well just set up the other hubs as well.

Reply:
Please read the documentation. I don't know how to make this any clearer. 

Connect locally. MAC address NOT required.
Connect via the cloud. MAC address IS required. This is to stop a duplicate instance in HA for local and cloud.

One hub, one login. Even the app from Genius Heating requires one unique login per hub. MAC address or no MAC address separate credentials are required for each hub. One login WILL NOT fetch all the hubs.

Reply:
Think about it for a while, to connect locally you need one unique IP address per hub. Multiple hubs CANNOT share the same IP address.

Reply:
Login from the cloud WILL NOT return the MAC address or local IP address.

Reply:
Check, that's what I wanted to know